src/main/java/tutoraid/storage/JsonTutorAidStudentStorage.java

```diff
@@ -22,37 +23,46 @@
     private static final Logger logger = LogsCenter.getLogger(JsonTutorAidStudentStorage.class);
 
     private Path filePath;
+    private ReadOnlyLessonBook lessonBook;
 
-    public JsonTutorAidStudentStorage(Path filePath) {
+    /**
+     * Constructor for the student storage object.
+     *
+     * @param filePath Path to the JSON file
+     * @param lessonBook Lesson book to create the Lesson-Student dependency
+     */
+    public JsonTutorAidStudentStorage(Path filePath, ReadOnlyLessonBook lessonBook) {
         this.filePath = filePath;
+        this.lessonBook = lessonBook;
     }
 
     public Path getStudentBookFilePath() {
         return filePath;
     }
 
     @Override
-    public Optional<ReadOnlyStudentBook> readStudentBook() throws DataConversionException {
-        return readStudentBook(filePath);
+    public Optional<ReadOnlyStudentBook> readStudentBook(ReadOnlyLessonBook lessonBook) throws DataConversionException {
+        return readStudentBook(filePath, lessonBook);
     }
 
     /**
-     * Similar to {@link #readStudentBook()}.
+     * Similar to {@link #readStudentBook(ReadOnlyLessonBook)}.
      *
      * @param filePath location of the data. Cannot be null.
      * @throws DataConversionException if the file is not in the correct format.
      */
-    public Optional<ReadOnlyStudentBook> readStudentBook(Path filePath) throws DataConversionException {
+    public Optional<ReadOnlyStudentBook> readStudentBook(Path filePath, ReadOnlyLessonBook lessonBook)
+            throws DataConversionException {
         requireNonNull(filePath);
 
         Optional<JsonSerializableStudentBook> jsonStudentBook = JsonUtil.readJsonFile(
                 filePath, JsonSerializableStudentBook.class);
-        if (!jsonStudentBook.isPresent()) {
+        if (jsonStudentBook.isEmpty()) {
             return Optional.empty();
```

Comment:
A proposed fix to solve the bug where `Alex Yeoh` is not showing up under the `Students` field for the `Maths 1` lesson when the app is first launched with sample data
```suggestion
            try {
                return Optional.of(new JsonSerializableStudentBook(SampleDataUtil.getSampleStudentBook(lessonBook))
                        .toModelType(lessonBook));
            } catch (IllegalValueException ive) {
                logger.info("Illegal values found in " + filePath + ": " + ive.getMessage());
                throw new DataConversionException(ive);
            }
```
This will ensure that `JsonSerializableStudentBook#toModelType()` is called even when there is initially no student book in the hard disk